cpp/core/jni/JniCommon.h

```diff
@@ -235,9 +235,9 @@ static inline std::string JStringToCString(JNIEnv* env, jstring string) {
   int32_t jlen, clen;
   clen = env->GetStringUTFLength(string);
   jlen = env->GetStringLength(string);
-  std::vector<char> buffer(clen);
-  env->GetStringUTFRegion(string, 0, jlen, buffer.data());
-  return std::string(buffer.data(), clen);
+  char buffer[clen];
```

Comment:
clen maybe too large to explode the stack?

Reply:
You can see the stacktrace in https://github.com/oap-project/gluten/pull/604

Reply:
The root cause is the large memory allocation (>5G), we shouldn't use stack memory to bypass it here?
Instead we should report OOM error.

Reply:
Here we do not allocate 5G, it core dump when throw OOM exception, after this fix, it can throw OOM error.
The clen is just the error message length which should be small.